py/objstr.c

```diff
@@ -832,6 +832,10 @@ STATIC bool arg_looks_numeric(mp_obj_t arg) {
     ;
 }
 
+STATIC bool arg_looks_string(mp_obj_t arg) {
+    return MP_OBJ_IS_TYPE(arg, &mp_type_str) || MP_OBJ_IS_STR(arg);
```

Comment:
Actually, I think https://docs.python.org/3/library/string.html#formatstrings table right after `The available string presentation types are` is the key. You can see, if it is `None` field, than it is same as `s`. And I don't get your example with inheritance, can you show some code?